### ISSUE TITLE
getAccessToken 

### ISSUE THREAD

**Issue by TLeiter** [OfficeDev/office-js#3759]
It seems the OfficeRuntime.auth.getAccessToken is not working in the spam-reporting context. I testes two similar code, one trigger by a taskpane button and another by the spam-reporting action. In the second case I always get SSO failed with "5001".

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 695f4ab9-5e6f-3b7a-04fe-3f7769958e98
* Version Independent ID: 695f4ab9-5e6f-3b7a-04fe-3f7769958e98
* Content: [Implement an integrated spam-reporting add-in (preview) - Office Add-ins](https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/spam-reporting)
* Content Source: [docs/outlook/spam-reporting.md](https://github.com/OfficeDev/office-js-docs-pr/blob/main/docs/outlook/spam-reporting.md)
* Service: **outlook**
* Sub-service: **add-ins**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

**Comment by AlexJerabek** [comment]
Thanks for raising this question @TLeiter.

@davidchesnut, could you and @samantharamon investigate this issue with regard to the spam reporting scenario?

**Comment by timwan10** [comment]
The Spam Reporting / Report Phishing context has the same requirements as Event Based Add-ins:

https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/autolaunch?tabs=xmlmanifest#event-based-activation-behavior-and-limitations

Most importantly, you need to have a .well-known file, and make SURE it is accessible. (a good way to check is if you can just navigate to it in a web browser)

https://learn.microsoft.com/en-us/office/dev/add-ins/outlook/use-sso-in-event-based-activation

**Comment by TLeiter** [comment]
I do see a request to /.well-known/microsoft-officeaddins-allowed.json which contains the following entries:
```
{
    "allowed":
    [
        "https://localhost:3000/command/command.js",
        "https://localhost:3000/command/command.html",
    ]
}
```
The URLs are allowed in the "singe page application" configuration tab in Azure:

The manifest contains the "WebApplicationInfo" section:
```
 <WebApplicationInfo>
        <Id>[redacted]</Id>
        <Resource>api://localhost:3000/[redacted]</Resource>
        <Scopes>
          <Scope>openid</Scope>
          <Scope>profile</Scope>
          <Scope>email</Scope>
          <Scope>User.Read</Scope>
          <Scope>Mail.Read</Scope>
          <Scope>Mail.ReadWrite</Scope>
          <Scope>Mail.Send</Scope>
        </Scopes>
      </WebApplicationInfo>
```
IN addition here is the code which call getAccessToken:

```
const defaultSSO = {
    allowSignInPrompt: false,
    allowConsentPrompt: false,
};

function onSpamReport(event) {
 

    const options = JSON.parse(JSON.stringify(defaultSSO));

    // Begin promise chain.
    return OfficeRuntime.auth.getAccessToken(options).then((accessToken) =>
    {
        console.log('getAccessToken done');
        console.log('accesstoken: ' + accessToken); //<- we never reach this point

       ...
```


In the log file, I get the following entries:

```
20/09/2023 15:13:59 Verbose Runtime [Console] [Log] SDX Control is ready!       
20/09/2023 15:13:59 Verbose Runtime [Console] [Log] onSpamReport        
20/09/2023 15:13:59 Verbose Runtime [Console] [Log] MailboxHostExecuteApi invoked!      
20/09/2023 15:13:59 Verbose Runtime [Console] [Log] RunLaunchEventHandlerFunctionWithData was invoked with funcName = onSpamReport          
20/09/2023 15:13:59 Verbose Runtime [Console] [Log] callWebServerAPI(POST, https://localhost:3000/report_phishing)      
20/09/2023 15:13:59 Verbose Runtime [Console] [Log] MailboxHostExecuteApi invoked!      
20/09/2023 15:13:59 Monitorable WebApplicationInfo  SSO failed with "5001" error code
```

**Comment by timwan10** [comment]
Can you include the ReportPhishingCommandSurface extension point part of your interface? What source location is it pointing to (and what are the resource values?) 

Do you have a LaunchEvents section in the manifest too?

(or really if you can provide the manifest + .js file you are using that would be helpful)

**Comment by Adrian-MSFT** [comment]
@TLeiter can you confirm that you can access the well-known file from your browser?

Also, these steps are also required: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/use-sso-to-get-office-signed-in-user-token?tabs=vs2019.

I went through these steps myself and was able to get a successful response

**Comment by TLeiter** [comment]
> @TLeiter can you confirm that you can access the well-known file from your browser?
> 
> Also, these steps are also required: https://learn.microsoft.com/en-us/office/dev/add-ins/develop/use-sso-to-get-office-signed-in-user-token?tabs=vs2019.
> 
> I went through these steps myself and was able to get a successful response

Yes the well-known file is accessible and I do see a request to fetch it coming from Outlook when it launch.

**Comment by TLeiter** [comment]
[](url)> Can you include the ReportPhishingCommandSurface extension point part of your interface? What source location is it pointing to (and what are the resource values?)
> 
> Do you have a LaunchEvents section in the manifest too?
> 
> (or really if you can provide the manifest + .js file you are using that would be [helpful)](url)

Here is the manifest file (with some part redacted)

```<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<OfficeApp xmlns="http://schemas.microsoft.com/office/appforoffice/1.1"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xmlns:bt="http://schemas.microsoft.com/office/officeappbasictypes/1.0"
  xmlns:mailappor="http://schemas.microsoft.com/office/mailappversionoverrides/1.0" xsi:type="MailApp">
  <Id>[REDACTED</Id>
  <Version>1.0.0.1</Version>
  <ProviderName>[REDACTED]</ProviderName>
  <DefaultLocale>en-US</DefaultLocale>
  <DisplayName DefaultValue="[REDACTED]"/>
  <Description DefaultValue="Repport phishing email"/>
  <IconUrl DefaultValue="https://localhost:3000/assets/icon64.png"/>
  <HighResolutionIconUrl DefaultValue="https://localhost:3000/assets/icon128.png"/>
  <SupportUrl DefaultValue="[REDACTED]"/>
  <AppDomains>
    <AppDomain>https://localhost:3000</AppDomain>
  </AppDomains>
  <Hosts>
    <Host Name="Mailbox"/>
  </Hosts>
  <Requirements>
    <Sets>
      <Set Name="Mailbox" MinVersion="1.1"/>
    </Sets>
  </Requirements>
  <FormSettings>
    <Form xsi:type="ItemRead">
      <DesktopSettings>
        <SourceLocation DefaultValue="https://localhost:3000/taskpane/taskpane.html"/>
        <RequestedHeight>250</RequestedHeight>
      </DesktopSettings>
    </Form>
  </FormSettings>
  <Permissions>ReadWriteMailbox</Permissions>
  <Rule xsi:type="RuleCollection" Mode="Or">
    <Rule xsi:type="ItemIs" ItemType="Message" FormType="Read"/>
    <Rule xsi:type="ItemIs" ItemType="Appointment" FormType="Read"/>
  </Rule>
  <DisableEntityHighlighting>false</DisableEntityHighlighting>
  <VersionOverrides xmlns="http://schemas.microsoft.com/office/mailappversionoverrides" xsi:type="VersionOverridesV1_0">
    <VersionOverrides xmlns="http://schemas.microsoft.com/office/mailappversionoverrides/1.1" xsi:type="VersionOverridesV1_1">
      <Requirements>
        <bt:Sets DefaultMinVersion="1.13">
          <bt:Set Name="Mailbox"/>
        </bt:Sets>
      </Requirements>
      <Hosts>
        <Host xsi:type="MailHost">
          <Runtimes>
            <Runtime resid="WebViewRuntime.Url">
              <!-- References the JavaScript file that contains the spam-reporting event handler. This is used by Outlook on Windows. -->
              <Override type="javascript" resid="JSRuntime.Url"/>
            </Runtime>
          </Runtimes>
          <DesktopFormFactor>
            <FunctionFile resid="WebViewRuntime.Url" />
            <ExtensionPoint xsi:type="ReportPhishingCommandSurface">
              <ReportPhishingCustomization>
                <!-- Configures the ribbon button. -->
                <Control xsi:type="Button" id="spamReportingButton">
                  <Label resid="spamButton.Label"/>
                  <Supertip>
                    <Title resid="spamButton.Label"/>
                    <Description resid="spamSuperTip.Text"/>
                  </Supertip>
                  <Icon>
                    <bt:Image size="16" resid="Icon.16x16"/>
                    <bt:Image size="32" resid="Icon.32x32"/>
                    <bt:Image size="80" resid="Icon.80x80"/>
                  </Icon>
                  <Action xsi:type="ExecuteFunction">
                    <FunctionName>onSpamReport</FunctionName>
                  </Action>
                </Control>
                <!-- Configures the preprocessing dialog. -->
                <PreProcessingDialog>
                  <Title resid="PreProcessingDialog.Label"/>
                  <Description resid="PreProcessingDialog.Text"/>
                  <ReportingOptions>
                    <Title resid="OptionsTitle.Label"/>
                    <Option resid="Option1.Label"/>
                    <Option resid="Option2.Label"/>
                    <Option resid="Option3.Label"/>
                  </ReportingOptions>
                  <FreeTextLabel resid="FreeText.Label"/>
                  <MoreInfo>
                    <MoreInfoText resid="MoreInfo.Label"/>
                    <MoreInfoUrl resid="MoreInfo.Url"/>
                  </MoreInfo>
                </PreProcessingDialog>
              <!-- Identifies the runtime to be used. This is also referenced by the Runtime element. -->
                <SourceLocation resid="WebViewRuntime.Url"/>
              </ReportPhishingCustomization> 
            </ExtensionPoint> 
          </DesktopFormFactor>
        </Host>
      </Hosts>
      <Resources>
        <bt:Images>
          <bt:Image id="Icon.16x16" DefaultValue="https://localhost:3000/assets/icon16.png"/>
          <bt:Image id="Icon.32x32" DefaultValue="https://localhost:3000/assets/icon32.png"/>
          <bt:Image id="Icon.80x80" DefaultValue="https://localhost:3000/assets/icon80.png"/>
        </bt:Images>
        <bt:Urls>
          <bt:Url id="WebViewRuntime.Url" DefaultValue="https://localhost:3000/command/command.html"/>
          <bt:Url id="JSRuntime.Url" DefaultValue="https://localhost:3000/command/command.js"/>
        </bt:Urls>
        <bt:ShortStrings>
          <bt:String id="spamButton.Label" DefaultValue="Report Spam Message"/>
          <bt:String id="PreProcessingDialog.Label" DefaultValue="Report Spam Message"/>
          <bt:String id="OptionsTitle.Label" DefaultValue="Why are you reporting this email?"/>
          <bt:String id="FreeText.Label" DefaultValue="Provide additional information, if any:"/>
          <bt:String id="MoreInfo.Label" DefaultValue="To learn more about reporting unsolicited messages, see "/>
          <bt:String id="Option1.Label" DefaultValue="Received spam email."/>
          <bt:String id="Option2.Label" DefaultValue="Received a phishing email."/>
          <bt:String id="Option3.Label" DefaultValue="I'm not sure this is a legitimate email."/>
        </bt:ShortStrings>
        <bt:LongStrings>
          <bt:String id="spamSuperTip.Text" DefaultValue="Report an unsolicited message."/>
          <bt:String id="PreProcessingDialog.Text" DefaultValue="Thank you for reporting this message."/>

          <bt:String id="ActionButton.Tooltip" DefaultValue="Perform an action when clicked."/>
        </bt:LongStrings>
      </Resources>
      <WebApplicationInfo>
        <Id>9ce5c57b-fb90-[REDACTED]</Id>
        <Resource>api://localhost:3000/9ce5c57b-fb90-[REDACTED]</Resource>
        <Scopes>
          <Scope>openid</Scope>
          <Scope>profile</Scope>
          <Scope>email</Scope>
          <!--Scope>User.Read</Scope> -->
          <!--<Scope>Mail.Read</Scope> -->
          <!--<Scope>Mail.ReadWrite</Scope> -->
          <!--<Scope>Mail.Send</Scope> -->
        </Scopes>
      </WebApplicationInfo>
      <ExtendedPermissions>
        <ExtendedPermission>AppendOnSend</ExtendedPermission>
      </ExtendedPermissions>
    </VersionOverrides>
  </VersionOverrides>
</OfficeApp>
```

**Comment by Adrian-MSFT** [comment]
@TLeiter, Thank you for providing the manifest. We were able to repro this issue and have opened a bug for this - Internal ID: 8443432.

In the meantime, as a work-around you can add a LaunchEvent extension point to your manifest, and that should enable the API to successfully be called:

```<OfficeApp xmlns="http://schemas.microsoft.com/office/appforoffice/1.1"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xmlns:bt="http://schemas.microsoft.com/office/officeappbasictypes/1.0"
  xmlns:mailappor="http://schemas.microsoft.com/office/mailappversionoverrides/1.0" xsi:type="MailApp">
  <Id>[REDACTED</Id>
  <Version>1.0.0.1</Version>
  <ProviderName>[REDACTED]</ProviderName>
  <DefaultLocale>en-US</DefaultLocale>
  <DisplayName DefaultValue="[REDACTED]"/>
  <Description DefaultValue="Repport phishing email"/>
  <IconUrl DefaultValue="https://localhost:3000/assets/icon64.png"/>
  <HighResolutionIconUrl DefaultValue="https://localhost:3000/assets/icon128.png"/>
  <SupportUrl DefaultValue="[REDACTED]"/>
  <AppDomains>
    <AppDomain>https://localhost:3000</AppDomain>
  </AppDomains>
  <Hosts>
    <Host Name="Mailbox"/>
  </Hosts>
  <Requirements>
    <Sets>
      <Set Name="Mailbox" MinVersion="1.1"/>
    </Sets>
  </Requirements>
  <FormSettings>
    <Form xsi:type="ItemRead">
      <DesktopSettings>
        <SourceLocation DefaultValue="https://localhost:3000/taskpane/taskpane.html"/>
        <RequestedHeight>250</RequestedHeight>
      </DesktopSettings>
    </Form>
  </FormSettings>
  <Permissions>ReadWriteMailbox</Permissions>
  <Rule xsi:type="RuleCollection" Mode="Or">
    <Rule xsi:type="ItemIs" ItemType="Message" FormType="Read"/>
    <Rule xsi:type="ItemIs" ItemType="Appointment" FormType="Read"/>
  </Rule>
  <DisableEntityHighlighting>false</DisableEntityHighlighting>
  <VersionOverrides xmlns="http://schemas.microsoft.com/office/mailappversionoverrides" xsi:type="VersionOverridesV1_0">
    <VersionOverrides xmlns="http://schemas.microsoft.com/office/mailappversionoverrides/1.1" xsi:type="VersionOverridesV1_1">
      <Requirements>
        <bt:Sets DefaultMinVersion="1.13">
          <bt:Set Name="Mailbox"/>
        </bt:Sets>
      </Requirements>
      <Hosts>
        <Host xsi:type="MailHost">
          <Runtimes>
            <Runtime resid="WebViewRuntime.Url">
              <!-- References the JavaScript file that contains the spam-reporting event handler. This is used by Outlook on Windows. -->
              <Override type="javascript" resid="JSRuntime.Url"/>
            </Runtime>
          </Runtimes>
          <DesktopFormFactor>
            <FunctionFile resid="WebViewRuntime.Url" />
            <ExtensionPoint xsi:type="ReportPhishingCommandSurface">
              <ReportPhishingCustomization>
                <!-- Configures the ribbon button. -->
                <Control xsi:type="Button" id="spamReportingButton">
                  <Label resid="spamButton.Label"/>
                  <Supertip>
                    <Title resid="spamButton.Label"/>
                    <Description resid="spamSuperTip.Text"/>
                  </Supertip>
                  <Icon>
                    <bt:Image size="16" resid="Icon.16x16"/>
                    <bt:Image size="32" resid="Icon.32x32"/>
                    <bt:Image size="80" resid="Icon.80x80"/>
                  </Icon>
                  <Action xsi:type="ExecuteFunction">
                    <FunctionName>onSpamReport</FunctionName>
                  </Action>
                </Control>
                <!-- Configures the preprocessing dialog. -->
                <PreProcessingDialog>
                  <Title resid="PreProcessingDialog.Label"/>
                  <Description resid="PreProcessingDialog.Text"/>
                  <ReportingOptions>
                    <Title resid="OptionsTitle.Label"/>
                    <Option resid="Option1.Label"/>
                    <Option resid="Option2.Label"/>
                    <Option resid="Option3.Label"/>
                  </ReportingOptions>
                  <FreeTextLabel resid="FreeText.Label"/>
                  <MoreInfo>
                    <MoreInfoText resid="MoreInfo.Label"/>
                    <MoreInfoUrl resid="MoreInfo.Url"/>
                  </MoreInfo>
                </PreProcessingDialog>
              <!-- Identifies the runtime to be used. This is also referenced by the Runtime element. -->
                <SourceLocation resid="WebViewRuntime.Url"/>
              </ReportPhishingCustomization> 
            </ExtensionPoint>
            <ExtensionPoint xsi:type="LaunchEvent">
              <LaunchEvents>
                <LaunchEvent Type="OnNewMessageCompose" FunctionName="onNewMessageComposeHandler"/>
              </LaunchEvents>
              <SourceLocation resid="WebViewRuntime.Url"/>
            </ExtensionPoint>
          </DesktopFormFactor>
        </Host>
      </Hosts>
      <Resources>
        <bt:Images>
          <bt:Image id="Icon.16x16" DefaultValue="https://localhost:3000/assets/icon16.png"/>
          <bt:Image id="Icon.32x32" DefaultValue="https://localhost:3000/assets/icon32.png"/>
          <bt:Image id="Icon.80x80" DefaultValue="https://localhost:3000/assets/icon80.png"/>
        </bt:Images>
        <bt:Urls>
          <bt:Url id="WebViewRuntime.Url" DefaultValue="https://localhost:3000/command/command.html"/>
          <bt:Url id="JSRuntime.Url" DefaultValue="https://localhost:3000/command/command.js"/>
        </bt:Urls>
        <bt:ShortStrings>
          <bt:String id="spamButton.Label" DefaultValue="Report Spam Message"/>
          <bt:String id="PreProcessingDialog.Label" DefaultValue="Report Spam Message"/>
          <bt:String id="OptionsTitle.Label" DefaultValue="Why are you reporting this email?"/>
          <bt:String id="FreeText.Label" DefaultValue="Provide additional information, if any:"/>
          <bt:String id="MoreInfo.Label" DefaultValue="To learn more about reporting unsolicited messages, see "/>
          <bt:String id="Option1.Label" DefaultValue="Received spam email."/>
          <bt:String id="Option2.Label" DefaultValue="Received a phishing email."/>
          <bt:String id="Option3.Label" DefaultValue="I'm not sure this is a legitimate email."/>
        </bt:ShortStrings>
        <bt:LongStrings>
          <bt:String id="spamSuperTip.Text" DefaultValue="Report an unsolicited message."/>
          <bt:String id="PreProcessingDialog.Text" DefaultValue="Thank you for reporting this message."/>

          <bt:String id="ActionButton.Tooltip" DefaultValue="Perform an action when clicked."/>
        </bt:LongStrings>
      </Resources>
      <WebApplicationInfo>
        <Id>9ce5c57b-fb90-[REDACTED]</Id>
        <Resource>api://localhost:3000/9ce5c57b-fb90-[REDACTED]</Resource>
        <Scopes>
          <Scope>openid</Scope>
          <Scope>profile</Scope>
          <Scope>email</Scope>
          <!--Scope>User.Read</Scope> -->
          <!--<Scope>Mail.Read</Scope> -->
          <!--<Scope>Mail.ReadWrite</Scope> -->
          <!--<Scope>Mail.Send</Scope> -->
        </Scopes>
      </WebApplicationInfo>
      <ExtendedPermissions>
        <ExtendedPermission>AppendOnSend</ExtendedPermission>
      </ExtendedPermissions>
    </VersionOverrides>
  </VersionOverrides>
</OfficeApp>

**Comment by timwan10** [comment]
(Note that the SourceLocation in the LaunchEvents section AND the ReportPhishingCustomization sections MUST be the same value. (This will be true even after the bug is fixed)) 

For right now, the bug happens if ONLY ReportPhishingCustomization and NOT LaunchEvents are included. When a fixed build is released, ReportPhishingCustomization can be used on it's own.

**Comment by TLeiter** [comment]
@Adrian-MSFT I still have the error even if I add the LaunchEvents section in my manifest.  I guess I wil wait till the bug is fixed to continue my tests.

**Comment by Adrian-MSFT** [comment]
@TLeiter, thank you for your help tracking down this bug - I was not able to reproduce this with the launchevent section in the manifest. Can you please provide your manifest again (containing the launchevent section) so I can test with it?

**Comment by TLeiter** [comment]
@Adrian-MSFT sorry it was a mistake, it is working with the lauchevent section but not  without it.

**Comment by Adrian-MSFT** [comment]
@TLeiter this should be resolved in builds 16.0.17018 and greater. Thank you again for your bug report and help.
